WordPress/Classes/ViewRelated/Reader/ReaderSearchViewController.swift

```diff
@@ -0,0 +1,164 @@
+    }
+
+
+
+    // MARK: - Configuration
+
+
+    func setupSearchBar() {
+        // Appearance must be set before the search bar is added to the view hierarchy.
+        let placeholderText = NSLocalizedString("Search on WordPress.com", comment: "Placeholder text for the Reader search feature.")
+        let attributes = WPStyleGuide.defaultSearchBarTextAttributes(WPStyleGuide.grey())
+        let attributedPlaceholder = NSAttributedString(string: placeholderText, attributes: attributes)
+        UITextField.appearanceWhenContainedInInstancesOfClasses([UISearchBar.self, ReaderSearchViewController.self]).attributedPlaceholder = attributedPlaceholder
+        let textAttributes = WPStyleGuide.defaultSearchBarTextAttributes(WPStyleGuide.greyDarken30())
+        UITextField.appearanceWhenContainedInInstancesOfClasses([UISearchBar.self, ReaderSearchViewController.self]).defaultTextAttributes = textAttributes
+
+        searchBar.autocapitalizationType = .None
+        searchBar.translucent = false
+        searchBar.tintColor = WPStyleGuide.grey()
+        searchBar.barTintColor = WPStyleGuide.greyLighten30()
+        searchBar.backgroundImage = UIImage()
+        searchBar.setImage(UIImage(named: "icon-clear-textfield"), forSearchBarIcon: .Clear, state: .Normal)
+
+        let size = searchBarSearchIconSize * UIScreen.mainScreen().scale
+        let image = Gridicon.iconOfType(GridiconType.Search, withSize: CGSize(width: size, height: size))
+
+        searchBar.setImage(image, forSearchBarIcon: .Search, state: .Normal)
+        searchBar.accessibilityIdentifier = "Search"
+    }
+
+
+    func configureLabel() {
+        let text = NSLocalizedString("What would you like to find?", comment: "A short message that is a call to action for the Reader's Search feature.")
+        let attributes = WPNUXUtility.titleAttributesWithColor(WPStyleGuide.greyDarken20()) as! [String: AnyObject]
+        label.attributedText = NSAttributedString(string: text, attributes: attributes)
+    }
+
+
+    func configureBackgroundTapRecognizer() {
+        backgroundTapRecognizer = UITapGestureRecognizer(target: self, action: #selector(ReaderSearchViewController.handleBackgroundTap(_:)))
+        backgroundTapRecognizer.cancelsTouchesInView = true
+        backgroundTapRecognizer.enabled = false
+        view.addGestureRecognizer(backgroundTapRecognizer)
+    }
+
+
+    // MARK: - Actions
+
+
+    func endSearch() {
+        searchBar.resignFirstResponder()
+    }
+
+
+    /// Constructs a ReaderSearchTopic from the search phrase and sets the
+    /// embedded stream to the topic.
+    ///
+    func performSearch() {
+        guard let phrase = searchBar.text else {
+            return
+        }
+
+        let context = ContextManager.sharedInstance().mainContext
+        let service = ReaderTopicService(managedObjectContext: context)
+
+        let topic = service.searchTopicForSearchPhrase(phrase)
+        streamController.readerTopic = topic
```

Comment:
(Just a thought) maybe we should assert on `streamController` (or make sure it's not nil?)